### DISCUSSION TITLE
project: error in devtools::document()

### DISCUSSION THREAD

**Discussion by lynseythomson** [IBAHCM/RPiR#480] [Archive] [answered]
Hi,

I get this error after trying to run devtools::document():

devtools::document()
ℹ Updating lynseythomsonDiversity documentation
ℹ Loading lynseythomsonDiversity
Writing NAMESPACE
Writing NAMESPACE
Writing p1_species_richness-open-paren-close-paren.Rd
Warning message:
In setup_ns_exports(path, export_all, export_imports) :
  Objects listed as exports, but not present in namespace: p1_simpson_diversity()

I'm confused what this means. I've tried clearing everything and restarting R before running again but the same message!

Any help appreciated!
Thank you

**Comment by richardreeve** [comment, marked as answer]
@lynseythomson - you seem to have a function whose name is `p1_species_richness()`. That might look right at first glance, but actually the function's name is just `p1_species_richness`. Hopefully you can see the difference? You can see what the documentation for a function should be here - https://github.com/IBAHCM/soniamitchellBCI/blob/master/R/sample_by_species.R - for instance. You don't need the function documentation to build into a notebook, so the information there is enough. If you are still getting errors after fixing this, you may need to delete the `man` folder before rerunning `document()` to make sure that you are starting from scratch.

Note also that I provided you with a new function `RPiR::assert_no_globals()` - see #443 - so that you don't need to use all of the `library(codetools)` and other stuff I showed you earlier in the course.

**Reply by lynseythomson** [reply]
oh thank you for pointing that out! I took out the () after the function name, ran document/install again and it worked!

I'll check out the `RPiR::assert_no_globals()` and try tidy up my code.

Thank you for the help!
